WordPress/Classes/ViewRelated/Blog/SiteManagement/SiteTagsViewController.swift

```diff
@@ -0,0 +1,266 @@
+        self.blog = blog
+        super.init(style: .grouped)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setAccessibilityIdentifier()
+        applyStyleGuide()
+        applyTitle()
+        setupTable()
+        setupNavigationBar()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        refreshResultsController(predicate: defaultPredicate)
+        refreshTags()
+        refreshNoResultsView()
+    }
+
+    private func setAccessibilityIdentifier() {
+        tableView.accessibilityIdentifier = TableConstants.accesibilityIdentifier
+    }
+
+    private func applyStyleGuide() {
+        WPStyleGuide.configureColors(for: view, andTableView: tableView)
+        WPStyleGuide.configureAutomaticHeightRows(for: tableView)
+    }
+
+    private func applyTitle() {
+        title = NSLocalizedString("Tags", comment: "Label for the Tags Section in the Blog Settings")
+    }
+
+    private func setupTable() {
+        tableView.tableFooterView = UIView(frame: .zero)
+        let nibName = UINib(nibName: TableConstants.cellIdentifier, bundle: nil)
+        tableView.register(nibName, forCellReuseIdentifier: TableConstants.cellIdentifier)
+        setupRefreshControl()
+    }
+
+    private func setupRefreshControl() {
+        let control = UIRefreshControl()
+        control.addTarget(self, action: #selector(refreshTags), for: .valueChanged)
+        refreshControl = control
+    }
+
+    @objc private func refreshResultsController(predicate: NSPredicate) {
+        resultsController.fetchRequest.predicate = predicate
+        resultsController.fetchRequest.sortDescriptors = sortDescriptors
+        do {
+            try resultsController.performFetch()
+
+            tableView.reloadData()
+        } catch {
+            tagsFailedLoading(error: error)
+        }
+    }
+
+    @objc private func refreshTags() {
+        let tagsService = PostTagService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        tagsService.syncTags(for: blog, success: { [weak self] tags in
+            self?.refreshControl?.endRefreshing()
+        }) { [weak self] error in
+            self?.tagsFailedLoading(error: error)
+        }
+    }
+
+    private func setupNavigationBar() {
+        configureRightButton()
+        configureSearchBar()
+    }
+
+    private func configureRightButton() {
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add,
+                                                            target: self,
+                                                            action: #selector(createTag))
+    }
+
+    private func configureSearchBar() {
+       tableView.tableHeaderView = searchController.searchBar
+    }
+
+    @objc private func createTag() {
+        let emptyTag = PostTag()
```

Comment:
Guessing this is just stubbed for now?

Reply:
Yes, I was planning on dealing with this in the next PR